utils.go

```diff
@@ -1,8 +1,13 @@
 package rsmt2d
 
 func flattenChunks(chunks [][]byte) []byte {
-	flattened := chunks[0]
-	for _, chunk := range chunks[1:] {
+	length := 0
+	for _, chunk := range chunks {
+		length += len(chunk)
+	}
```

Comment:
Aren't all these lengths the same? So you could read e.g. the first chunk, take its `len`  (let's call that `firstChunkLen`)  and then do `make([]byte, 0, len(chunks)*firstChunkLen)`? This probably doesn't have any big impact and it has the downside that it only works if chunks have the same size (which I thought would be expected where this method is used anyways though). I don't think it is worth exploring much. Just curious if you did though.

Reply:
I just tried and it makes literally zero performance difference. But losing the generality would be not-good, so suggest keeping it as is.

Reply:
Cool, thanks!